test/libsolidity/util/TestFileParser.cpp

```diff
@@ -0,0 +1,267 @@
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/util/TestFileParser.h>
+
+#include <test/Options.h>
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/throw_exception.hpp>
+#include <fstream>
+#include <memory>
+#include <stdexcept>
+
+using namespace dev;
+using namespace langutil;
+using namespace solidity;
+using namespace dev::solidity::test;
+using namespace std;
+
+namespace
+{
+	void expect(std::string::iterator& _it, std::string::iterator _end, std::string::value_type _c)
+	{
+		if (_it == _end || *_it != _c)
+			throw std::runtime_error(std::string("Invalid test expectation. Expected: \"") + _c + "\".");
+		++_it;
+	}
+
+	template<typename IteratorType>
+	void skipWhitespace(IteratorType& _it, IteratorType _end)
+	{
+		while (_it != _end && isspace(*_it))
+			++_it;
+	}
+
+	template<typename IteratorType>
+	void skipSlashes(IteratorType& _it, IteratorType _end)
+	{
+		while (_it != _end && *_it == '/')
+			++_it;
+	}
+}
+
+string TestFileParser::bytesToString(bytes const& _bytes, TestFileParser::ByteFormat const& _format)
+{
+	// TODO: Convert from compact big endian if padded
+	bytes byteRange{_bytes};
+	stringstream resultStream;
+	switch(_format.type)
+	{
+		case TestFileParser::ByteFormat::SignedDec:
+			if (*_bytes.begin() & 0x80)
+			{
+				for (auto& v: byteRange)
+					v ^= 0xFF;
+				resultStream << "-" << fromBigEndian<u256>(byteRange) + 1;
+			}
+			else
+				resultStream << fromBigEndian<u256>(byteRange);
+			break;
+		case TestFileParser::ByteFormat::UnsignedDec:
+			resultStream << fromBigEndian<u256>(byteRange);
+			break;
+	}
+	return resultStream.str();
+}
+
+pair<bytes, TestFileParser::ByteFormat> TestFileParser::stringToBytes(std::string _string)
+{
+	bytes result;
+	ByteFormat format;
+	auto it = _string.begin();
+	while (it != _string.end())
+	{
+		if (isdigit(*it) || (*it == '-' && (it + 1) != _string.end() && isdigit(*(it + 1))))
+		{
+			format.type = TestFileParser::ByteFormat::UnsignedDec;
+
+			bool isNegative = (*it == '-');
+			if (isNegative)
+				format.type = TestFileParser::ByteFormat::SignedDec;
+
+			auto valueBegin = it;
+			while (it != _string.end() && !isspace(*it) && *it != ',')
+				++it;
+
+			bytes newBytes;
+			u256 numberValue(std::string(valueBegin, it));
+			// TODO: Convert to compact big endian if padded
+			if (numberValue == u256(0))
+				newBytes = bytes{0};
+			else
+				newBytes = toBigEndian(numberValue);
+			result += newBytes;
+		}
+		else
+			BOOST_THROW_EXCEPTION(std::runtime_error("Test expectations contain invalidly formatted data."));
```

Comment:
solAssert?

Reply:
I think in testing we don't use `solAssert`. `BOOST_FAIL_MSG` or `BOOST_REQUIRE(false, "Test...")` would be more appropriate?